tools/shared/src/main/scala/org/scalajs/core/tools/linker/checker/IRChecker.scala

```diff
@@ -312,41 +316,47 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
           "but must be NoType")
     }
 
-    val bodyStats = body match {
-      case Block(stats) => stats
-      case _            => body :: Nil
-    }
+    body match {
+      case None =>
```

Comment:
Hum good point. I think the intention in the style guide is that it applies to folds on *collections*. `Option.fold` has been treated differently in our codebase.